src/app/clusters/bindings/bindings.cpp

```diff
@@ -20,155 +20,265 @@
+                    .endpointId = Optional<EndpointId>(entry.remote),
+                    .clusterId  = entry.clusterId,
+                };
+                ReturnErrorOnFailure(subEncoder.Encode(value));
+            }
+            else if (entry.type == EMBER_MULTICAST_BINDING)
+            {
+                Binding::Structs::TargetStruct::Type value = {
+                    .groupId   = Optional<GroupId>(entry.groupId),
+                    .clusterId = entry.clusterId,
+                };
+                ReturnErrorOnFailure(subEncoder.Encode(value));
+            }
         }
-    }
-
-    emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
-    return true;
+        return CHIP_NO_ERROR;
+    });
 }
 
-bool emberAfBindingClusterUnbindCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
-                                         const Commands::Unbind::DecodableType & commandData)
+CHIP_ERROR BindingTableAccess::Write(const ConcreteDataAttributePath & path, AttributeValueDecoder & decoder)
 {
-    NodeId nodeId             = commandData.nodeId;
-    GroupId groupId           = commandData.groupId;
-    ClusterId clusterId       = commandData.clusterId;
-    EndpointId remoteEndpoint = commandData.endpointId;
-    EndpointId localEndpoint  = commandPath.mEndpointId;
-    FabricIndex fabricIndex   = commandObj->GetAccessingFabricIndex();
-    EmberBindingTableEntry bindingEntry;
-
-    ChipLogDetail(Zcl, "RX: UnbindCallback");
-
-    if ((groupId != 0 && nodeId != 0) || (groupId == 0 && nodeId == 0))
-    {
-        ChipLogError(Zcl, "Binding: Invalid request");
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_MALFORMED_COMMAND);
-        return true;
-    }
-    if (groupId)
-    {
-        bindingEntry = EmberBindingTableEntry::ForGroup(fabricIndex, groupId, localEndpoint, clusterId);
-    }
-    else
+    switch (path.mAttributeId)
     {
-        bindingEntry = EmberBindingTableEntry::ForNode(fabricIndex, nodeId, localEndpoint, remoteEndpoint, clusterId);
+    case Binding::Attributes::BindingList::Id:
+        return WriteBindingTable(path, decoder);
+    default:
+        break;
     }
+    return CHIP_NO_ERROR;
+}
 
-    uint8_t bindingIndex;
-    if (getBindingIndex(bindingEntry, &bindingIndex) != EMBER_SUCCESS)
+CHIP_ERROR BindingTableAccess::WriteBindingTable(const ConcreteDataAttributePath & path, AttributeValueDecoder & decoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    if (!path.IsListOperation() || path.mListOp == ConcreteDataAttributePath::ListOperation::ReplaceAll)
     {
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_NOT_FOUND);
-        return true;
-    }
+        printf("replace all\n");
+        DecodableBindingListType newBindingList;
+
+        ReturnErrorOnFailure(decoder.Decode(newBindingList));
+        ReturnErrorOnFailure(CheckValidBindingList(newBindingList));
+
+        // Remove entries not in the new binding list
```

Comment:
This should only be removing entries that have the same fabric index as the incoming interaction.  Fabrics should not be deleting each other's bindings.